xml/hwbp_registers.xml

```diff
@@ -549,6 +552,37 @@
         </field>
     </register>
 
+    <register name="External Trigger" short="exttrigger" address="0x7a1">
+        This register is accessible as \RcsrTdataOne when \FcsrTdataOneType is 7.
+
+        This trigger fires when a selected external debug trigger input
+        asserts (e.g. when a pulse is received).  One of up to 16 external
+        debug trigger inputs (coming from other blocks within the hart,
+        e.g. signaling an hpmcounter overflow) can be selected.  Hardware
+        may support none or just a few external debug trigger inputs
+        (starting with external debug trigger input 0 and continuing
+        sequentially).  Unsupported inputs are hardwired to be inactive.
+
+        <field name="type" bits="XLEN-1:XLEN-4" access="R" reset="4" />
+        <field name="dmode" bits="XLEN-5" access="WARL" reset="0" />
+        <field name="hit" bits="XLEN-6" access="WARL" reset="0">
+            If this bit is implemented, the hardware sets it when this
+            trigger matches. The trigger's user can set or clear it at any
+            time. It is used to determine which
+            trigger(s) matched.  If the bit is not implemented, it is always 0
+            and writing it has no effect.
+        </field>
+        <field name="0" bits="XLEN-7:10" access="R" reset="0" />
+        <field name="select" bits="9:6" access="WARL" reset="0">
+            Selects one of up to 16 external debug trigger inputs that would
```

Comment:
Would it be better to make this a 16-bit field, allowing the trigger to fire when any of the selected inputs sends a signal?

Reply:
Sounds good.  The downside is that there are fewer bits for other purposes but I don't think there's a need for a large number of additional bits.